### PR TITLE
Create routing policy for empty deployment spec MERGEOK

### PR DESCRIPTION
**Pull request by mpolden**
Manual deployments can have a empty deployment spec, without any instances. This
change ensures that we create a routing policy and update DNS for such
deployments.

### CODE REVIEW

**Review by oyving — APPROVED** [review]

**Comment by jonmv** [comment]
Will this help when application has submitted a spec with instance "a", then deploys to `dev` for the first time under a new instance "b"?

**Comment by jonmv** [comment]
I took the liberty of adding a test with this scenario which fails.

**Comment by mpolden** [comment]
Instances.......................!

@jonmv PTAL.

**Review by jonmv — APPROVED** [review]
A question, but probably OK :)

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/RoutingPolicies.java [hunk 1]
```diff
@@ -187,29 +183,39 @@ private void removeObsoleteEndpointsFromDns(AllocatedLoadBalancers loadBalancers
...
+            this.deploymentSpec = deploymentSpec;
+        }
+
+        /** Compute all endpoint IDs for given load balancer */
+        private Set<EndpointId> endpointIdsOf(LoadBalancer loadBalancer) {
+            if (zone.environment().isManuallyDeployed()) { // Manual deployments do not have any configurable endpoints
+                return Set.of();
+            }
+            var instanceSpec = deploymentSpec.instance(loadBalancer.application().instance());
+            if (instanceSpec.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/RoutingPolicies.java]: It's an `Optional`.

[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/RoutingPolicies.java]: Yes. When would it not be present? All automatic deployments will be for instances that are given in the deployment spec. All others should be covered above.

[the remark this answers is not in the document]
Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/RoutingPolicies.java]: Sure, I can make it throw instead (in future PR).

**Comment by mpolden** [comment]
Java build passed.

**Pull request merged**